src/components/Contacts/ContactDetails/ContactDontationsTab/PartnershipInfo/EditPartnershipInfoModal/EditPartnershipInfoModal.tsx

```diff
@@ -0,0 +1,431 @@
+                              {t(
+                                'Commitments can only be set if status is Partner - Financial',
+                              )}
+                            </Typography>
+                          }
+                        >
+                          <InfoIcon />
+                        </Tooltip>
+                      ) : (
+                        <ArrowDropDownIcon />
+                      )
+                    }
+                  >
+                    {Object.values(PledgeFrequencyEnum).map((value) => (
+                      <MenuItem key={value} value={value}>
+                        {t(value)}
+                      </MenuItem>
+                    ))}
+                  </Select>
+                </FormControl>
+              </ContactInputWrapper>
+              <ContactInputWrapper>
+                <KeyboardDatePicker
+                  onChange={(date): void =>
+                    setFieldValue('pledgeStartDate', date)
+                  }
+                  value={
+                    pledgeStartDate
+                      ? DateTime.fromISO(pledgeStartDate).toLocaleString()
+                      : null
+                  }
+                  format="MM/dd/yyyy"
+                  clearable
+                  label={t('Start Date')}
+                  inputProps={{ 'aria-label': t('Start Date') }}
+                  fullWidth
+                  KeyboardButtonProps={{
+                    'aria-label': 'change start date',
+                  }}
+                />
+              </ContactInputWrapper>
+              <ContactInputWrapper>
+                <KeyboardDatePicker
+                  onChange={(date) =>
+                    !date ? null : setFieldValue('nextAsk', date)
+                  }
+                  value={
+                    nextAsk ? DateTime.fromISO(nextAsk).toLocaleString() : null
+                  }
+                  format="MM/dd/yyyy"
+                  clearable
+                  label={t('Next Ask Increase')}
+                  inputProps={{ 'aria-label': t('Next Ask Increase') }}
+                  fullWidth
+                  KeyboardButtonProps={{
+                    'aria-label': 'change next ask date',
+                  }}
+                />
+              </ContactInputWrapper>
+              <ContactInputWrapper>
+                <CheckboxLabel
+                  control={
+                    <Checkbox
+                      checked={pledgeReceived}
+                      onChange={() =>
+                        setFieldValue('pledgeReceived', !pledgeReceived)
+                      }
+                    />
+                  }
+                  label={t('Commitment Recieved')}
+                />
+              </ContactInputWrapper>
+              <ContactInputWrapper>
+                <CheckboxLabel
+                  control={
+                    <Checkbox
+                      checked={!!!noAppeals}
```

Comment:
This is interesting logic? Wouldn't !noAppeals be the same? or is that some javascript magic I don't know which is probable 😄

Reply:
When I read your comment here I started getting really confused trying to think through this, but I remembered why I did this! ```noAppeals``` can be ```null```, so the double bang or ```!!``` can convert non-booleans to be booleans. So ```!!null``` would be ```false```.  And then it gets a little weird. So it's a strange situation, the value is ```noAppeals```, but the UI is ```Send Appeals```, so you have to do the opposite of ```noAppeals```. So let's say ```noAppeals``` was false, meaning the UI would need to display ```Send Appeals``` as checked, or ```true```. Adding ```!!``` would make it stay ```false``` or would convert ```null``` to ```false```. But the UI needs it to be true,  so we add one more bang to get ```!!!``` and that makes it ```true```. Confusing...yes, but it works haha!

Reply:
I think this is the same as
```suggestion
                      checked={!noAppeals}
```
The first bang should return a boolean and invert the value. Everything after just inverts it since it's already a boolean.

Reply:
Ah, so I was overcomplicating it haha. Thanks, I'll change that!